dvc/scm/git/backend/gitpython.py

```diff
@@ -215,7 +215,17 @@ def list_tags(self):
         return [t.name for t in self.repo.tags]
 
     def list_all_commits(self):
-        return [c.hexsha for c in self.repo.iter_commits("--all")]
```

Comment:
`git rev-parse --all` returns every commit reachable by any ref in `refs/*`. If this is actually the behavior we want, we need to adjust this to be `git rev-parse --exclude=<experiment_refs> --all`. But it seems to me like the actual desired behavior for `--all-commits` is commits reachable from "standard" git refs (`HEAD`, `refs/tags`, `refs/heads` (branches), `remotes/*/refs/tags`, `remotes/*/refs/heads`).

Unless there's actually DVC users out there that are using their own custom refspecs, this change should not affect anything.